From b7f98d1d9930b15f07342f67d2560aecd4ebc401 Mon Sep 17 00:00:00 2001
From: Fomalhaut Weisszwerg <weisszwerg@gmail.com>
Date: Mon, 27 Nov 2023 21:24:57 +0900
Subject: [PATCH 077/188] set Cargo feature resolver to v2 (#8917)

* fix: version of Cargo feature resolver.

This commit solve the ambiguity to determin the version of resolver.
To get more detail, see the following two documents:

- https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions
- https://doc.rust-lang.org/edition-guide/rust-2021/default-cargo-resolver.html

* unified: Rust edition in all workspaces.

Now, the Rust 2021 is available in all workspaces.

* fined up: Cargo.toml by using workspace inheritance.

To get more detail of the `workspace.package` table, see a following document:

- https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table
---
 Cargo.toml              | 6 ++++++
 helix-core/Cargo.toml   | 5 +++--
 helix-dap/Cargo.toml    | 5 +++--
 helix-event/Cargo.toml  | 5 +++--
 helix-loader/Cargo.toml | 5 +++--
 helix-lsp/Cargo.toml    | 5 +++--
 helix-parsec/Cargo.toml | 5 +++--
 helix-term/Cargo.toml   | 6 +++---
 helix-tui/Cargo.toml    | 5 +++--
 helix-vcs/Cargo.toml    | 5 +++--
 helix-view/Cargo.toml   | 5 +++--
 11 files changed, 36 insertions(+), 21 deletions(-)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,4 +1,5 @@
 [workspace]
+resolver = "2"
 members = [
   "helix-core",
   "helix-view",
@@ -37,3 +38,8 @@ package.helix-term.opt-level = 2
 [workspace.dependencies]
 tree-sitter = { version = "0.20", git = "https://github.com/tree-sitter/tree-sitter", rev = "ab09ae20d640711174b8da8a654f6b3dec93da1a" }
 nucleo = "0.2.0"
+
+[workspace.package]
+edition = "2021"
+license = "MPL-2.0"
+rust-version = "1.70"
--- a/helix-core/Cargo.toml
+++ b/helix-core/Cargo.toml
@@ -2,8 +2,9 @@
 name = "helix-core"
 version = "0.6.0"
 authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
-edition = "2021"
-license = "MPL-2.0"
+edition.workspace = true
+license.workspace = true
+rust-version.workspace = true
 description = "Helix editor core editing primitives"
 categories = ["editor"]
 repository = "https://github.com/helix-editor/helix"
--- a/helix-dap/Cargo.toml
+++ b/helix-dap/Cargo.toml
@@ -2,8 +2,9 @@
 name = "helix-dap"
 version = "0.6.0"
 authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
-edition = "2018"
-license = "MPL-2.0"
+edition.workspace = true
+license.workspace = true
+rust-version.workspace = true
 description = "DAP client implementation for Helix project"
 categories = ["editor"]
 repository = "https://github.com/helix-editor/helix"
--- a/helix-event/Cargo.toml
+++ b/helix-event/Cargo.toml
@@ -2,8 +2,9 @@
 name = "helix-event"
 version = "0.6.0"
 authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
-edition = "2021"
-license = "MPL-2.0"
+edition.workspace = true
+license.workspace = true
+rust-version.workspace = true
 categories = ["editor"]
 repository = "https://github.com/helix-editor/helix"
 homepage = "https://helix-editor.com"
--- a/helix-loader/Cargo.toml
+++ b/helix-loader/Cargo.toml
@@ -3,8 +3,9 @@ name = "helix-loader"
 version = "0.6.0"
 description = "A post-modern text editor."
 authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
-edition = "2021"
-license = "MPL-2.0"
+edition.workspace = true
+license.workspace = true
+rust-version.workspace = true
 categories = ["editor"]
 repository = "https://github.com/helix-editor/helix"
 homepage = "https://helix-editor.com"
--- a/helix-lsp/Cargo.toml
+++ b/helix-lsp/Cargo.toml
@@ -2,8 +2,9 @@
 name = "helix-lsp"
 version = "0.6.0"
 authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
-edition = "2021"
-license = "MPL-2.0"
+edition.workspace = true
+license.workspace = true
+rust-version.workspace = true
 description = "LSP client implementation for Helix project"
 categories = ["editor"]
 repository = "https://github.com/helix-editor/helix"
--- a/helix-parsec/Cargo.toml
+++ b/helix-parsec/Cargo.toml
@@ -2,8 +2,9 @@
 name = "helix-parsec"
 version = "0.6.0"
 authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
-edition = "2021"
-license = "MPL-2.0"
+edition.workspace = true
+license.workspace = true
+rust-version.workspace = true
 description = "Parser combinators for Helix"
 categories = ["editor"]
 repository = "https://github.com/helix-editor/helix"
--- a/helix-term/Cargo.toml
+++ b/helix-term/Cargo.toml
@@ -3,14 +3,14 @@ name = "helix-term"
 version = "0.6.0"
 description = "A post-modern text editor."
 authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
-edition = "2021"
-license = "MPL-2.0"
+edition.workspace = true
+license.workspace = true
 categories = ["editor", "command-line-utilities"]
 repository = "https://github.com/helix-editor/helix"
 homepage = "https://helix-editor.com"
 include = ["src/**/*", "README.md"]
 default-run = "hx"
-rust-version = "1.70"
+rust-version.workspace = true
 
 [features]
 default = ["git"]
--- a/helix-tui/Cargo.toml
+++ b/helix-tui/Cargo.toml
@@ -5,8 +5,9 @@ authors = ["Blaž Hrastnik <blaz@mxxn.io
 description = """
 A library to build rich terminal user interfaces or dashboards
 """
-edition = "2021"
-license = "MPL-2.0"
+edition.workspace = true
+license.workspace = true
+rust-version.workspace = true
 categories = ["editor"]
 repository = "https://github.com/helix-editor/helix"
 homepage = "https://helix-editor.com"
--- a/helix-vcs/Cargo.toml
+++ b/helix-vcs/Cargo.toml
@@ -2,8 +2,9 @@
 name = "helix-vcs"
 version = "0.6.0"
 authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
-edition = "2021"
-license = "MPL-2.0"
+edition.workspace = true
+license.workspace = true
+rust-version.workspace = true
 categories = ["editor"]
 repository = "https://github.com/helix-editor/helix"
 homepage = "https://helix-editor.com"
--- a/helix-view/Cargo.toml
+++ b/helix-view/Cargo.toml
@@ -2,8 +2,9 @@
 name = "helix-view"
 version = "0.6.0"
 authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
-edition = "2021"
-license = "MPL-2.0"
+edition.workspace = true
+license.workspace = true
+rust-version.workspace = true
 description = "UI abstractions for use in backends"
 categories = ["editor"]
 repository = "https://github.com/helix-editor/helix"
